tests/phpunit/tests/db.php

```diff
@@ -2472,4 +2472,91 @@ public function data_parse_db_host() {
 			),
 		);
 	}
+
+	/**
+	 * @ticket 55414
+	 */
+	public function test_suppress_query_error() {
+		global $wpdb;
+
+		$old_show_errors     = $wpdb->show_errors;
+		$old_suppress_errors = $wpdb->suppress_errors;
+		$old_last_query      = $wpdb->last_query;
+		$old_error_log_path  = ini_get( 'error_log' );
+
+		$suppress_query_error_filter = new MockAction();
+		$log_query_error_action      = new MockAction();
+
+		try {
+			// Redirect error logging so this test doesn't print messages in the test output.
+			$temp_error_log = tmpfile();
+			$this->assertNotEmpty( $temp_error_log, 'Unable to create temp file for error logging' );
+			$temp_error_log_meta = stream_get_meta_data( $temp_error_log );
+			ini_set( 'error_log', $temp_error_log_meta['uri'] );
+
+			// Disable `show_errors` to avoid being labeled a risky test.
+			$wpdb->show_errors = false;
+			$wpdb->last_query  = 'expected-query';
+
+			add_filter( 'suppress_query_error', array( $suppress_query_error_filter, 'filter' ), 10, 3 );
+			add_filter( 'log_query_error', array( $log_query_error_action, 'action' ), 10 );
+
+			// Test that the current suppress_errors setting is passed to filter.
+			$wpdb->suppress_errors = false;
+			$wpdb->print_error( 'expected-error' );
+			$this->assertSame( 1, $suppress_query_error_filter->get_call_count() );
```

Comment:
> All PHPUnit assertions, as well as all WordPress custom assertions, allow for a $message parameter to be passed. This message will be displayed when the assertion fails and can help immensely when debugging a test. **This parameter should always be used if more than one assertion is used in a test method**
> [Ref](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions)

(my emphasis)

🔢 Applies to other assertions in the PR.

Reply:
This is a good idea (and also fixes a handbook violation). Fixed.